core/src/indexer/process.rs

```diff
@@ -561,19 +547,17 @@ async fn handle_logs_result(
                 tokio::time::sleep(Duration::from_millis(1000)).await;
             }
 
-            if !fn_data.is_empty() {
-                return if config.index_event_in_order() {
+            // Important that we call this for every event even if there are no logs.
+            // This is because we need to sync the last seen block
+            if config.index_event_in_order() {
+                trigger_event(config, fn_data, result.to_block).await;
```

Comment:
we did this on purpose we did not want to have to write a db event up if nothing has changed to overload the db each time - we have a global last seen which is good but i dont think we want to trigger an event with no logs at all?